compiler/sem/semfold.nim

```diff
@@ -378,8 +379,16 @@ proc evalOp*(m: TMagic, n, a, b, c: PNode; idgen: IdGenerator; g: ModuleGraph):
     result = copyTree(a)
     result.typ = n.typ
   of mEqProc:
-    result = newIntNodeT(toInt128(ord(
-        exprStructuralEquivalentStrictSym(a, b))), n, idgen, g)
+    g.config.internalAssert(a.kind in {nkSym, nkNilLit} and
+                            b.kind in {nkSym, nkNilLit},
+                            n.info, "mEqProc: invalid AST")
+    let isEqual = if a.kind != b.kind:
+                    false
+                  elif a.kind == nkSym: # and b.kind == nkSym:
+                    a.sym == b.sym
+                  else: # a.kind == nkNilLit and b.kind == nkNilLit
+                    true
```

Comment:
```suggestion
    let isEqual =
      if a.kind != b.kind:  false
      elif a.kind == nkSym: a.sym == b.sym # b.kind == nkSym
      else:                 true # a.kind == b.kind == nkNilLit
```
Soft suggestion regarding code style (recently introduced multi-line if/case/etc. expression tend to use this formatting).